ibis/omniscidb/operations.py

```diff
@@ -313,6 +313,74 @@ def extract_field_formatter(translator, expr, sql_attr=sql_attr):
     return extract_field_formatter
 
 
+def _n_weeks_year(year: ir.NumericValue):
+    """
+    Return the number of the total of weeks for the given year.
+
+    Parameters
+    ----------
+    year : ibis.expr.types.NumericValue
+
+    Returns
+    -------
+    Ibis.expr.types.Expr
+
+    See Also
+    --------
+    https://en.wikipedia.org/wiki/ISO_week_date
+
+    """
+    # adjustment factor for week year extraction
+    def adjustment_factor(year):
+        return (year + (year // 4) - (year // 100) + (year // 400)) % 7
+
+    need_adjustment = (adjustment_factor(year) == 4) | (
+        adjustment_factor(year - 1) == 3
+    )
+
+    return 52 + (ibis.case().when(need_adjustment, 1).else_(0).end())
+
+
+def _woy_preliminary(d: Union[ir.DateValue, ir.TimestampValue]) -> ir.Expr:
+    """
+    Return a preliminary week of year (WOY).
+
+    Parameters
+    ----------
+    d : ibis.expr.types.DateValue or ibis.expr.types.TimestampValue
+
+    Returns
+    -------
+    ir.Expr
+
+    See Also
+    --------
+    https://en.wikipedia.org/wiki/ISO_week_date
+    """
+    result = d.day_of_year() - d.day_of_week.index() + 11
+    return (result // 7).cast('int16')
+
+
+def _extract_woy(translator, expr) -> str:
```

Comment:
I think being explicit about names will make things easier. In this PR `woy` is clear what it means by the context. But when working on something unrelated, going to read the docstring will be needed. While naming it `week_of_year` will make it immediate to know.

```suggestion
def _extract_week_of_year(translator, expr) -> str:
```